File: src/main/java/com/blooddonation/system/models/donations/BloodDonation/IBloodStock.java
package com.blooddonation.system.models.donations.BloodDonation;
import com.blooddonation.system.models.donations.BloodDonation.Beneficiaries.Beneficiary;
import jakarta.persistence.Embeddable;

public interface IBloodStock {
    void registerBeneficiary(Beneficiary beneficiary);
    void removeBeneficiary(Beneficiary beneficiary);
    void notifyBeneficiaries(BloodTypeEnum bloodType, int newAmount);
}

File: models/blood_donations/IBloodStock.php
<?php
// File: IBloodStock.php

interface IBloodStock
{
    public function addBeneficiary(IBeneficiary $beneficiary): void;
    public function removeBeneficiary(IBeneficiary $beneficiary): void;
    public function notifyBeneficiaries(DonationType $bloodDonationType, BloodTypeEnum $bloodType, float $amount): void;
}
